pkg/servejschmidt/servejschmidt.go

```diff
@@ -0,0 +1,176 @@
+package servejschmidt
+
+import (
+	"fmt"
+	"log"
+	"net/http"
+	"net/url"
+	"strings"
+
+	"github.com/Contrast-Security-OSS/go-test-bench/internal/common"
+	"github.com/Contrast-Security-OSS/go-test-bench/internal/injection/cmdi"
+	"github.com/Contrast-Security-OSS/go-test-bench/internal/injection/sqli"
+	"github.com/Contrast-Security-OSS/go-test-bench/internal/pathtraversal"
+	"github.com/Contrast-Security-OSS/go-test-bench/internal/ssrf"
+	"github.com/Contrast-Security-OSS/go-test-bench/internal/unvalidated"
+	"github.com/Contrast-Security-OSS/go-test-bench/internal/xss"
+	"github.com/julienschmidt/httprouter"
+)
+
+// Pd is unchanging parameter data shared between all routes.
+var Pd = common.ConstParams{
+	Year:      2022,
+	Logo:      "https://blog.golang.org/gopher/header.jpg",
+	Framework: "Julienschmidt",
+}
+
+func newHandler(v common.Route) func(w http.ResponseWriter, r *http.Request, p httprouter.Params) {
+	return func(w http.ResponseWriter, r *http.Request, p httprouter.Params) {
+		log.Println(v.Name, r.URL.Path)
+
+		// Split off the first element from elems, which should be the mode.
+		mode := strings.Split(strings.Trim(p.ByName("elems"), "/"), "/")[0]
+		for _, s := range v.Sinks {
+			if p.ByName("sink") != s.URL {
+				continue
+			}
+			mode := common.Safety(mode)
+			switch mode {
+			case common.NOOP, common.Safe, common.Unsafe:
+				// valid modes
+			default:
+				// invalid
+				w.WriteHeader(http.StatusNotFound)
+				return
+			}
+
+			in := common.GetUserInput(r)
+			data, mime, status := s.Handler(mode, in, httpHandlerPair{w, r})
+			if len(data) == 0 {
+				// don't unconditionally write response, as it can result in
+				// - an error log of "http: superfluous response.WriteHeader call"
+				return
+			}
+			w.WriteHeader(status)
+			if len(mime) == 0 {
+				mime = "text/plain"
+			}
+			w.Header().Set("Content-Type", mime)
+			w.Header().Set("Cache-Control", "no-store") //makes development a whole lot easier
+			fmt.Fprint(w, data)
+			return
+		}
+		// does not match any sink or the main page
+		w.WriteHeader(http.StatusNotFound)
+	}
+}
+
+func add(router *httprouter.Router, rt common.Route) {
+	for _, s := range rt.Sinks {
+		if s.Handler == nil {
+			var err error
+			s.Handler, err = common.GenericHandler(s)
+			if err != nil {
+				log.Fatal(err)
+			}
+		}
+	}
+	// main page
+	router.GET(rt.Base+"/", func(w http.ResponseWriter, r *http.Request, _ httprouter.Params) {
+		log.Println(rt.Name, r.URL.Path)
+		var parms = common.Parameters{
+			ConstParams: Pd,
+			Name:        rt.Base,
+		}
+		err := common.Templates[rt.TmplFile].ExecuteTemplate(w, "layout.gohtml", &parms)
+		if err != nil {
+			log.Print(err.Error())
+			fmt.Fprintf(w, "template error: %s", err)
+		}
+	})
+
+	// Julienshmidt only allows one request method for each route,
+	// so each input method has to be a separate route.
+
+	router.GET(rt.Base+"/:sink/query/*elems", newHandler(rt))
```

Comment:
Agreed on iterating over `rt.Inputs`. Not sure what you mean by iterating over sinks here. Do you mean to add a separate handler logic for each sink? The only reason I didn't have it all in one handler like the other frameworks was due to julienschmidt handlers only working for exactly one request type. I'm not sure there's any other way to correctly match the request URLs we want to handler. I also like the fact that the path has both named parameters and catchall parameters, in case a go agent change breaks one or both of those down the line.